webpack-subresource-integrity/plugin.ts

```diff
@@ -240,29 +234,28 @@ more information."
         );
 
         if (childChunk.id !== null) {
-          this.hashByChunkId.set(childChunk.id, integrity);
+          this.hashByPlaceholder.set(
+            makePlaceholder(this.options.hashFuncNames, childChunk.id),
+            integrity
+          );
         }
         this.updateAssetIntegrity(sourcePath, integrity);
-        this.compilation.updateAsset(
-          sourcePath,
-          (x) => x,
-          (assetInfo) => {
-            if (!assetInfo) {
-              return undefined;
-            }
-
-            this.warnAboutLongTermCaching(assetInfo);
-
-            return {
-              ...assetInfo,
-              contenthash: Array.isArray(assetInfo.contenthash)
-                ? [...new Set([...assetInfo.contenthash, integrity])]
-                : assetInfo.contenthash
-                ? [assetInfo.contenthash, integrity]
-                : integrity,
-            };
+        this.compilation.updateAsset(sourcePath, newAsset, (assetInfo) => {
```

Comment:
Main change here is to pass the new source into `compilation.updateAsset()` method rather than setting it as a side-effect in `replaceAsset()`